giraf/giraf.go

```diff
@@ -17,6 +17,7 @@ type Giraf struct {
 	Flag      uint8
 	PosStrand bool
 	Path      *Path
+	Cigar     []cigar.ByteCigar
```

Comment:
Does this hold the same info as Aln, minus the base sequence of inserted bases?  I think this is here to be more efficient.  Should we change cigar.Cigar to be more efficient instead of have duplicate copies?  Maybe we could have Cigar as []cigar.ByteCigar and then a slice of dna that is the list of inserted sequences, such that if the ByteCigar had two insertions (a 10bp insertion and then later on a 5bp insertion), there would be a single slice (of length 15) holding those bases, which could be left as nil if those were not included.

Reply:
If it were up to me, I'd prefer to live in a world without fancy cigar all together (no []dna.Base whatsoever).

Reply:
Yes, I am aware that you are not a fan of the fancy cigar.  I think we should still continue to explore compression techniques/encodings that will allow us to reduce the amount of data that needs to be stored.  One relationship that has held true in the past is that the cost of sequencing has decreased faster than I/O speeds have increased.  This has made compression increasingly important (e.g. bam -> cram).  While we can gain a lot from binary encoding of the data and general compression with gzip, minimizing the amount of data we need to store is another path forward (that can be combined with the other two paths).  Most reads will be perfect matches to the graph (increasingly so as the graph grows), so we don't need to store the sequence for them since the cigar implicitly encodes it.  The only exception to the cigar implicitly encoding the sequence is insertions, which is why if we can find a compromise to encode these inserted sequences, we can not store the actual sequence of the read.

Reply:
I'll throw in my 2 cents here. I do quite like the fancy cigar, primarily for the enhanced compression that we can achieve as Craig mentions. That said, I agree with Eric that we should drop the Fancy Cigar in the human readable version of Giraf. During code review it seems that people were quite adamant that we keep the seq field in the human readable version which in my opinion undercuts the major utility of the fancy cigar. As-is we only stand to increase file size by incorporating the Fancy Cigar. I have had some discussions with Eric about this and my understanding is that we would like the human readable Giraf to have a sequence field and a semi-fancy byte cigar (encodes mismatches but not sequence). However, for the binary Giraf would would use a fancy cigar and a few other tricks to improve compression (read name prefixes + run-length encoding qual scores).

Reply:
i guess these discussions are unsettling to me because it feels like people think the alignment problem is solved. Even with my recent improvements, aligning reads to a full graph (stickleback) still proves to be difficult. I think it'll be hard to convince anyone to use the giraf format if we can't output an informative alignment.